src/MainWindow.cpp

```diff
@@ -1205,28 +1213,38 @@ void MainWindow::executeQuery()
                 if(query_part_type != StatementType::AttachStatement && query_part_type != StatementType::DetachStatement)
                     modified = true;
 
-                statusMessage = tr("Query executed successfully: %1 (took %2ms%3)").arg(queryPart.trimmed()).arg(timer.elapsed()).arg(stmtHasChangedDatabase);
+                statusMessage = tr("query executed successfully. Took %1ms%2").arg(timer.elapsed()).arg(stmtHasChangedDatabase);
                 ok = true;
                 break;
             }
             case SQLITE_MISUSE:
                 continue;
             default:
-                statusMessage = QString::fromUtf8(sqlite3_errmsg(pDb.get())) + ": " + queryPart;
-                ok = true;
+                ok = false;
                 break;
             }
             timer.restart();
         } else {
-            statusMessage = QString::fromUtf8(sqlite3_errmsg(pDb.get())) + ": " + queryPart;
-            sqlWidget->getEditor()->setErrorIndicator(execution_start_line, execution_start_index, execution_start_line, execution_end_index);
             ok = false;
 
         }
+        editor->lineIndexFromPosition(execution_start_position+1, &execution_start_line, &execution_start_index);
```

Comment:
I think the ```+1``` here is wrong.

If I execute ```xyz``` it underlines only the ```yz```. If I execute ```select 1;xyz``` the same happens. If I execute ```select 1; xyz``` (with a space after the semicolon) it correctly underlines the ```xyz```.

If I remove the ```+1``` it correctly underlines the ```xyz``` in the first case and in the second case. In the third case it underlines ``` xyz``` (that's with the space now) but it kind of makes sense because right after the semicolon the next/erroneous statement started.

With line breaks instead of spaces you don't see it except in the first case.

Reply:
I don't remember how I got to that `+1` but it might be related to the whitespace before the query that may be added to the error indicator. I'll take a look at it and remove the `+1` or even adjust the indicators taken into account the possibility of whitespace.

Reply:
I removed the `+1` for the reasons you explained, but I made a little adjustment when the start is at the end of the line. In that case it is adjusted to the beginning of the next line, so lines referenced in the log are the actual lines where the query starts.

I implemented the other suggestions too, so I will merge this now. Thanks for the review!